src/esockd_transport.erl

```diff
@@ -221,13 +221,25 @@ setopts(#proxy_socket{socket = Socket}, Opts) ->
       -> {ok, [{inet:stat_option(), integer()}]} | {error, inet:posix()}).
 getstat(Sock, Stats) when is_port(Sock) ->
     inet:getstat(Sock, Stats);
-getstat(#ssl_socket{tcp = Sock}, Stats) ->
+getstat(#ssl_socket{tcp = Sock}, Stats) when is_port(Sock) ->
     inet:getstat(Sock, Stats);
+getstat(#ssl_socket{ssl = SslSock}, Stats) ->
+    ssl_getstat(SslSock, Stats);
 getstat(SslSock = #sslsocket{}, Stats) ->
-    ssl:getstat(SslSock, Stats);
+    ssl_getstat(SslSock, Stats);
 getstat(#proxy_socket{socket = Sock}, Stats) ->
     getstat(Sock, Stats).
 
+-if(?OTP_RELEASE >= 23).
```

Comment:
Add some comments here describing the reason why we need special handling for OTP <= 22.